server/middlewares/stats.js

```diff
@@ -41,22 +41,52 @@ const parseTxBodyOutAmount = (txBody) => {
   }
 }
 
-const incrCountersBy = (key, value) => {
-  const [year, month, day] = getSlicedDate()
-  client.incrby(`${key}:total`, value)
-  client.incrby(`${key}:monthly:${year}-${month}`, value)
-  client.incrby(`${key}:daily:${year}-${month}-${day}`, value)
+const initVisitor = (trackingID) => {
+  return ua({
+    tid: trackingID,
+    cid: 555, //google analytics anonymous visitor
+    strictCidFormat: false, //to accept cid integer instead of UUID v4 format
+  })
+}
+
+const trackPageView = (argsObj) => {
+  const visitor = initVisitor(backendConfig.ADALITE_GA_TRACKING_ID)
+  const args = {
+    dp: argsObj.path,
+    dt: argsObj.hostname,
+    dh: argsObj.title,
+  }
+
+  visitor.pageview(args).send()
+}
+
+const trackEvent = (argsObj) => {
+  const visitor = initVisitor(backendConfig.ADALITE_GA_TRACKING_ID)
+  const args = {
+    ec: argsObj.category,
+    ea: argsObj.action,
+    el: argsObj.label,
+    ev: argsObj.value,
+  }
+
+  visitor.event(args).send()
 }
 
 const trackVisits = (req, res, next) => {
   // 'x-forwarded-for' due to internal heroku routing
   const ip = req.headers['x-forwarded-for'] || req.connection.remoteAddress
-
   const mydevice = device(req.headers['user-agent'])
 
   if (!knownIps.has(ip) && !mydevice.is('bot')) {
-    knownIps.add(ip)
-    incrCountersBy('visits', 1)
+    try {
+      trackPageView({
+        path: '/',
```

Comment:
this function takes also `hostname` and `title` - I would either provide them explicitly here or set them explicitly (if needed) in the `trackPageView` function

Reply:
they are optional, what hostname does production have? and usually people specify the title of the page the user is on, what should I do here since its a single page?

Reply:
the hostname can be extracted from the request, I would not set the title